devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy

```diff
@@ -122,6 +129,8 @@ def parse(String description) {
 		} else if (reportsBatteryPercentage() && descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && zigbee.convertHexToInt(descMap?.attrId) == BATTERY_PERCENTAGE_REMAINING && descMap.value) {
 			def batteryLevel = zigbee.convertHexToInt(descMap.value)
 			batteryPercentageEventHandler(batteryLevel)
+		} else if (reportBatteryPercentage() && descMap?.clusterInt == CLUSTER_BATTERY_LEVEL && descMap.value) {
+		    batteryPercentageEventHandler(zigbee.convertHexToInt(descMap.value))
```

Comment:
zigbee.POWER_CONFIGURATION_CLUSTER and CLUSTER_BATTERY_LEVEL are the same (0x0001) clusters. I am not sure how Your device sends reports (is it using BATTERY_PERCENTAGE_REMAINING - 0x0021 attribute), but I think it is worth checking to avoid code duplication.

Reply:
Since I double checked with our embedded engineers, and it turned out a mistake. I'll fix the code duplication by removing the adding codes